mifosng-android/src/main/java/com/mifos/mifosxdroid/online/search/SearchFragment.java

```diff
@@ -106,6 +113,29 @@ public void showUserInterface() {
         rv_search.addOnItemTouchListener(new RecyclerItemClickListener(getActivity(), this));
         rv_search.setHasFixedSize(true);
         rv_search.setAdapter(searchAdapter);
+
+        et_search.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                String query = charSequence.toString();
+                if (!query.equals("")) {
+                    showInstantSearchedProgressBar(true);
+                    searchPresenter.searchInstantResources(query, resources, false);
```

Comment:
After discussion with @therajanmaurya , he mentioned that we should not perform API calls in the `onTextChanged` method since it burdens the server with unnecessary API Calls. 

You might want to check [debounce](https://www.androidhive.info/RxJava/rxjava-operators-buffer-debounce/. ) to achieve this.